generator/generator_tests/altitude_test.cpp

```diff
@@ -0,0 +1,232 @@
+{
+  vector<T> dv;
+  dv.clear();
+  for (Rounded3DPoint const & p : geom3D)
+    dv.push_back(T(p.point2D));
+  return dv;
+}
+
+TAltitudes ConvertToAltitudes(TRounded3DGeom const & geom3D)
+{
+  TAltitudes altitudes;
+  altitudes.clear();
+  for (Rounded3DPoint const & p : geom3D)
+    altitudes.push_back(p.altitude);
+  return altitudes;
+}
+
+void FillAltitudes(vector<TRounded3DGeom> const & roadFeatures,
+                   MockAltitudeGetter::TMockAltitudes & altitudes)
+{
+  for (TRounded3DGeom const & geom3D : roadFeatures)
+  {
+    vector<m2::PointI> const featureGeom = ConvertTo2DGeom<m2::PointI>(geom3D);
+    TAltitudes const featureAlt = ConvertToAltitudes(geom3D);
+    CHECK_EQUAL(featureGeom.size(), featureAlt.size(), ());
+    for (size_t i = 0; i < featureGeom.size(); ++i)
+    {
+      auto it = altitudes.find(featureGeom[i]);
+      if (it != altitudes.end())
+      {
+        CHECK_EQUAL(it->second, featureAlt[i], ("Point", it->first, "is set with two different altitudes."));
+        continue;
+      }
+      altitudes[featureGeom[i]] = featureAlt[i];
+    }
+  }
+}
+
+void BuildMwmWithoutAltitude(vector<TRounded3DGeom> const & roadFeatures, LocalCountryFile & country)
+{
+  TestMwmBuilder builder(country, feature::DataHeader::country);
+
+  for (TRounded3DGeom const & geom3D : roadFeatures)
+    builder.Add(TestStreet(ConvertTo2DGeom<m2::PointD>(geom3D), string(), string()));
+}
+
+void ReadAndTestAltitudeInfo(MwmValue const & mwmValue, string const & mwmPath, MockAltitudeGetter & altitudeGetter)
+{
+  AltitudeLoader const loader(mwmValue);
+
+  auto processor = [&altitudeGetter, &loader](FeatureType const & f, uint32_t const & id)
+  {
+    f.ParseGeometry(FeatureType::BEST_GEOMETRY);
+    size_t const pointsCount = f.GetPointsCount();
+    TAltitudes const altitudes = loader.GetAltitudes(id, pointsCount);
+
+    if (!routing::IsRoad(feature::TypesHolder(f)))
+    {
+      TEST(altitudes.empty(), ());
+      return;
+    }
+
+    TEST_EQUAL(altitudes.size(), pointsCount, ());
+
+    for (size_t i = 0; i < pointsCount; ++i)
+      TEST_EQUAL(altitudeGetter.GetAltitude(f.GetPoint(i)), altitudes[i], ("A wrong altitude"));
+  };
+  feature::ForEachFromDat(mwmPath, processor);
+}
+
+void TestAltitudeSection(vector<TRounded3DGeom> const & roadFeatures)
+{
+  classificator::Load();
+  Platform & platform = GetPlatform();
+  string const testDirFullPath = my::JoinFoldersToPath(platform.WritableDir(), kTestDir);
+
+  MY_SCOPE_GUARD(removeTmpDir, [&] ()
+  {
+    GetPlatform().RmDirRecursively(testDirFullPath);
+  });
+
+  // Building mwm without altitude section.
+  platform.MkDir(testDirFullPath);
+  LocalCountryFile country(testDirFullPath, CountryFile(kTestMwm), 1);
+  BuildMwmWithoutAltitude(roadFeatures, country);
+
+  // Creating MockAltitudeGetter.
+  MockAltitudeGetter::TMockAltitudes altitudes;
+  FillAltitudes(roadFeatures, altitudes);
+  MockAltitudeGetter altitudeGetter(move(altitudes));
```

Comment:
Why not to just create a method Put in MockAltitudeGetter, and pass it directly to FillAltitudes?